azure_arc_k8s_jumpstart/docs/aro_terraform.md

```diff
@@ -0,0 +1,232 @@
+    --address-prefixes 10.0.0.0/23 \
+    --service-endpoints Microsoft.ContainerRegistry
+    ```
+
+  * Add an empty subnet for the worker nodes.
+    ```bash
+    az network vnet subnet create \
+    --resource-group $RESOURCEGROUP \
+    --vnet-name arcarodemo-vnet \
+    --name worker-subnet \
+    --address-prefixes 10.0.2.0/23 \
+    --service-endpoints Microsoft.ContainerRegistry
+    ```
+  * [Disable subnet private endpoint policies](https://docs.microsoft.com/en-us/azure/private-link/disable-private-link-service-network-policy) on the master subnet. This is required to be able to connect and manage the cluster.
+    ```bash
+    az network vnet subnet update \
+    --name master-subnet \
+    --resource-group $RESOURCEGROUP \
+    --vnet-name arcarodemo-vnet \
+    --disable-private-link-service-network-policies true
+    ```
+  * Create the Openshift Container Cluster:
+    ```bash
+    az aro create \
+    --resource-group $RESOURCEGROUP \
+    --name $AROCLUSTER \
+    --vnet arcarodemo-vnet \
+    --master-subnet master-subnet \
+    --worker-subnet worker-subnet
+    ```
+    ![](../img/aro/image3.png)
+
+  * Finish Deploying ARO Cluster:<br>
+
+    <img src="../img/aro/image7.png" width="100%">
+
+    
+
+# Connecting to Azure Arc
+
+Now that you have a running ```aro``` cluster, lets connect the ARO cluster to Azure Arc by:<br>
+
+  * Install ```oc``` command-line in Cloud Shell.
+    ```bash
+    cd ~
+    wget https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/openshift-client-linux.tar.gz
+
+    mkdir openshift
+    tar -zxvf openshift-client-linux.tar.gz -C openshift
+    echo 'export PATH=$PATH:~/openshift' >> ~/.bashrc && source ~/.bashrc
+    ```
+  * Fetch ```oc``` admin credentials to log into Openshift cluster command line.
+    ```bash
+    adminUser=$(az aro list-credentials --name $AROCLUSTER --resource-group $RESOURCEGROUP --query kubeadminUsername -o tsv)
+    adminPassword=$(az aro list-credentials --name $AROCLUSTER --resource-group $RESOURCEGROUP --query kubeadminPassword -o tsv)
+    ```
+  *  Retrieve the API server's address:
+      ```bash
+      apiServer=$(az aro show -g $RESOURCEGROUP -n $AROCLUSTER --query apiserverProfile.url -o tsv)
+      ```
+    
+  * Log into via ```oc``` command line
+    ```bash
+    oc login $apiServer -u $adminUser -p $adminPassword
+    ```
+    Output:
+    ![](../img/aro/image4.png)
+  
+  * Verify ```kubeconfig``` was pupulated:
+    ```bash
+    kubectl get nodes -o wide
+    ```
+    Output:<br>
+    <img src="../img/aro/image5.png" width="80%">
+
+
+
+
+  * Login to previously created [***Service Principal***](#prerequisites) <br>
+    ```bash
+    az login --service-principal -u mySpnClientId -p mySpnClientSecret --tenant myTenantID
+    ```
+
+  * Deploy Arc binaries using Azure CLI:
+    ```bash
+    az connectedk8s connect -n $ARC -g $RESOURCEGROUP
+    ```
+    Output:<br>
+    
+    <img src="../img/aro/image6.png" width="80%">
+
+
```

Comment:
I would argue we need to automate this entire thing with a shell script that just needs to be uploaded to Cloud Shell. It makes the experience a lot smoother IMO.